src/ModelContextProtocol/Server/AIFunctionMcpServerPrompt.cs

```diff
@@ -136,6 +138,10 @@ private static AIFunctionFactoryOptions CreateAIFunctionFactoryOptions(
         Throw.IfNull(function);
 
         List<PromptArgument> args = [];
+        HashSet<string>? requiredProps = function.JsonSchema.TryGetProperty("required", out JsonElement required)
+            ? new(required.EnumerateArray().Select(p => p.GetString()!), StringComparer.Ordinal)
```

Comment:
Do we need to validate that required is an array? What happens if you call EnumerateArray on an element that's not an array?

Reply:
Given that these AI functions can only be produced by `AIFunctionFactory` the `required` keyword is guaranteed to be an array if present.

Reply:
>Given that these AI functions can only be produced by AIFunctionFactory

That's not the case:
https://github.com/modelcontextprotocol/csharp-sdk/blob/a896fa805579da74a59257af77b9664936149a3b/src/ModelContextProtocol/Server/McpServerPrompt.cs#L211-L214

Reply:
So, leave as-is presumably? We already hardcode other assumption in code, e.g. that `"properties"` contains an object.

Reply:
If the resulting error will be reasonable enough, sure.